### PR TITLE
Web doc

### PR DESCRIPTION
**Pull request by gitkuba**

### CODE REVIEW

File: web.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Web
+
+**Prerequisite knowledge**: 
+
+## Intro
+Web - short for World Wide Web.
+
+No prerequisite knowledge required. If you see this document, you are probably aware of what computers are and that they can communicate with each other via global network of computer networks, called the Internet. If so, you're all set to begin your journey with the web.
```
Comment (kvas-damian) [inline web.md]: Maybe we could link to HTML or Hypertext here?
Info about prerequisite knowledge can be moved to section above.
This description seems very indirect to, it doesn't tell what Web is.

File: web.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+## Classification / acceptance questions
+### Apprentice:
+- I can tell what World Wide Web is.
+- I can tell the difference betweend web and the internet.
```
Comment (kvas-damian) [inline web.md]: Internet should be capitalized

File: web.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- I know what a web browser is.
+- I know what a web server is.
```
Comment (kvas-damian) [inline web.md]: Maybe we could add a point about HTML and linking between documents?

Comment (kvas-damian) [inline web.md]: We could also mention about supported media types.

File: web.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+### Intermediate
+- I can explain how does the Web work.
+- I know what DNS is.
+- I can describe step by step what happens when I put `www.youtube.com` into the address bar and hit enter before I can see the results.
+- I can make the above happen for my own custom domain name, which includes:
```
Comment (kvas-damian) [inline web.md]: What do you think about adding a bulletpoint about `AJAX/XHR` or `dynamic update

File: web.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+## Classification / acceptance questions
+### Apprentice:
+- I can tell what World Wide Web is.
+- I can tell the difference betweend web and the internet.
+- I know what a web browser is.
+- I know what a web server is.
+
+### Intermediate
+- I can explain how does the Web work.
+- I know what DNS is.
```
Comment (kvas-damian) [inline web.md]: I think we could  move it to `Prerequisite knowledge` section for this level

Comment (kvas-damian) [reply web.md]: Also it'd be nice to link HTTP and HTTPS in `Prerequisite knowledge`

File: web.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+- I can describe step by step what happens when I put `www.youtube.com` into the address bar and hit enter before I can see the results.
+- I can make the above happen for my own custom domain name, which includes:
+  - regestering a domain name
+  - adding a DNS record, that will point the domain to a
+  - web server I've set up, which will serve some
+  - HTML files I uploaded.
+
+### Advanced
```
Comment (kvas-damian) [inline web.md]: Maybe we could mention "web caching" in browser or CDN here?

File: web.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+
+## Sources
```
Comment (kvas-damian) [inline web.md:31]: What do you think about adding an article from wikipedia: https://en.wikipedia.org/wiki/World_Wide_Web#Dynamic_updates_of_web_pages for dinosuars like me that prefer text over video? :)

File: web.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+# Web
+
+**Prerequisite knowledge**: 
+
+## Intro
+Web - short for World Wide Web.
```
Comment (bognix) [inline web.md:8]: Hmm I don't think Web is the short of World Wide Web. WWW is the abbreviation of World Wide Web.

File: web.md (hunk 1 above)
Comment (bognix) [inline web.md]: global network of computer networks - I don't think this sentence is right.

File: web.md (hunk 2 above)
Comment (bognix) [reply web.md]: between**d**

File: web.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+- I can tell the difference betweend web and the internet.
+- I know what a web browser is.
+- I know what a web server is.
+
+### Intermediate
+- I can explain how does the Web work.
+- I know what DNS is.
+- I can describe step by step what happens when I put `www.youtube.com` into the address bar and hit enter before I can see the results.
+- I can make the above happen for my own custom domain name, which includes:
+  - regestering a domain name
```
Comment (bognix) [inline web.md]: in my opinion this should be in advanced section.

Beginner - understands the concept of the web
Intermediate - knows about DNS resolving and understands the basics of communication between browser and server
Advanced - knows how to set up this environment on her own

File: web.md (hunk 7 above)
Comment (bognix) [reply web.md:31]: https://dyn.com/blog/dns-why-its-important-how-it-works/

File: web.md (hunk 8 above)
Comment (gitkuba) [reply web.md:8]: https://dictionary.cambridge.org/us/dictionary/english/world-wide-web?q=Web

File: web.md (hunk 1 above)
Comment (gitkuba) [reply web.md]: good point

Comment (gitkuba) [reply web.md]: what makes You think so?

Comment (gitkuba) [reply web.md]: I'd say this is exactly what internet is :)

Comment (bognix) [reply web.md]: I meant in english

Comment (gitkuba) [reply web.md]: oh, allright :D How about `...communicate with each other via global system of interconnected computer networks, called the Internet`?

File: web.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+### Apprentice:
+- I can tell what World Wide Web is.
+- I can tell the difference betweend web and the internet.
+- I know what a web browser is.
+- I know what a web server is.
+
+### Intermediate
+- I can explain how does the Web work.
+- I know what DNS is.
+- I can describe step by step what happens when I put `www.youtube.com` into the address bar and hit enter before I can see the results.
```
Comment (rogatty) [inline web.md]: This is wide open for interpretation. Step by step might mean basic stuff like DNS -> request to youtube servers -> response but you can also go deep in what exactly happens on youtube servers - which you won't know unless you work there, and then what happens in the browser rendering engine. I'd make this point a bit more specific.

File: web.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+- I can make the above happen for my own custom domain name, which includes:
+  - regestering a domain name
+  - adding a DNS record, that will point the domain to a
+  - web server I've set up, which will serve some
```
Comment (rogatty) [inline web.md]: Let's either make it a sentence or bullet points, this mix doesn't read well.

File: web.md (hunk 3 above)
Comment (gitkuba) [reply web.md]: 🤔Uhm, you mean media supported by HTML5?  Specific formats like mp3? Response mimetypes supported by browsers? I think, I need an additional hint on that one ✋🏻

**Comment by gitkuba** [comment]
TODO: Name the links

**Comment by gitkuba** [comment]
TODO: Decide on where to put sources. Blocked by #20

File: web.md (hunk 3 above)
Comment (kvas-damian) [reply web.md]: I was thinking about listing "images, video, and audio". This gives you an overall view on web capabilities but if you need more info you can dive into details like formats etc.

File: web.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+**Where to look for answers**:
+- https://en.wikipedia.org/wiki/World_Wide_Web
+- https://www.youtube.com/watch?v=J8hzJxb0rpc
+
+### Intermediate aka _I know what's going on in the web_
+- I can explain how does the Web work.
+- I know what DNS, HTTP and HTML are.
+- I can describe step by step what happens when I put `www.youtube.com` into the address bar and hit enter before I can see the results. (Level of detail really depends on You. The more specific the answer, the better. But let's say naming 5 - 7 key steps  is enough to pass that question. Describing what exactly happens on youtube servers would be an overkill :D)
+
+**Prerequisite knowledge**: DNS, HTTP/HTTPS, HTML
```
Comment (kvas-damian) [inline web.md]: I think it's important to know what is required from you before you start answering questions. Maybe we could move this line directly under the level name?

File: web.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+- I can tell the difference between the Web and the Internet.
+- I know what a web browser is.
+- I know what a web server is.
+
+**Where to look for answers**:
+- https://en.wikipedia.org/wiki/World_Wide_Web
+- https://www.youtube.com/watch?v=J8hzJxb0rpc
+
+### Intermediate aka _I know what's going on in the web_
+- I can explain how does the Web work.
```
Comment (bognix) [inline web.md]: this one is the same as the first question from Apprentice level:
```
- I can explain how does the Web work and what are its most fundamental building blocks.
```

**Review by kvas-damian — APPROVED** [review]

File: web.md [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+- I can describe step by step what happens when I put `www.youtube.com` into the address bar and hit enter before I can see the results. (Level of detail really depends on You. The more specific the answer, the better. But let's say naming 5 - 7 key steps  is enough to pass that question. Describing what exactly happens on youtube servers would be an overkill :D)
+
+### Advanced aka _I can do some web myself_
+- I can register a domain and add a DNS record, that will point the domain to a web server.
+- I can set up a webserver that will serve HTML documents, images.
+- I can make practical use of AJAX/XHR to build a dynamic web application.
+
+### Master aka _I eat web for breakfast_
+- I am aware of common efficiency issues that come with a web application, and I know how to deal with them. (eg. caching, code optimazation, transfer speed optimization, CDN, load balancing)
+- ...
```
Comment (rogatty) [inline web.md:29]: ...?

**Pull request merged**